File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/Strategy.java
package Auxiliary_Modules;

public class Strategy {
    private final UniqueResponder allStrategies;
    private final int strategiesLim;
    private final UniqueItemSizeLimitedPriorityQueue activeStrategy;

    // Constructor
    public Strategy(UniqueResponder allStrategies, int strategiesLim) {
        this.allStrategies = allStrategies;
        this.strategiesLim = strategiesLim;
        this.activeStrategy = new UniqueItemSizeLimitedPriorityQueue();
        this.activeStrategy.setLimit(strategiesLim);
        for (int i = 0; i < this.strategiesLim; i++) {
            this.activeStrategy.add(this.allStrategies.getAResponse());
        }
    }

    // Evolve strategies
    public void evolveStrategies() {
        for (int i = 0; i < this.strategiesLim; i++) {
            this.activeStrategy.add(this.allStrategies.getAResponse());
        }
    }

    // Get strategy
    public String getStrategy() {
        return this.activeStrategy.getRNDElement();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/Strategy.kt
package auxiliary_modules

class Strategy(private val allStrategies: UniqueResponder, private val strategiesLim: Int) {
    private val activeStrategy: UniqueItemSizeLimitedPriorityQueue

    // Constructor
    init {
        activeStrategy = UniqueItemSizeLimitedPriorityQueue()
        activeStrategy.limit = strategiesLim
        for (i in 0 until strategiesLim) {
            activeStrategy.add(allStrategies.getAResponse())
        }
    }

    // Evolve strategies
    fun evolveStrategies() {
        for (i in 0 until strategiesLim) {
            activeStrategy.add(allStrategies.getAResponse())
        }
    }

    val strategy: String
        // Get strategy
        get() = activeStrategy.rNDElement
}